Add did method registration PR template (#360)
w3c/did-spec-registries: .github/PULL_REQUEST_TEMPLATE/register_did_method.md

## Register a DID Method

### Required Registration Criteria

Please confirm:

- [ ] The DID Method specification defines the identifer syntax.
- [ ] The DID Method specification defines the supported DID Operations.
- [ ] The DID Method specification contains a Privacy and Security Considerations.

### Optional Registration Criteria

Consider ensuring:

- [ ] The registration entry contains contact information.
- [ ] The registration entry contains registry information.

### Registration Process

In order to register a New DID Method, you must add a json file to [./methods](./methods).

Here is an example registration entry:

```json
{
  "status": "registered",
  "name": "example",
  "specification": "https://w3c-ccg.github.io/did-spec/",

>   "contact": "W3C Credentials Community Group",

Reviewer: ```suggestion
  "contactName": "W3C Credentials Community Group",
  "contactEmail": "public-credentials@w3.org",
  "contactWebsite": "https://w3c-ccg.github.io/",
```

>   "registry": "DID Specification"

Reviewer: ```suggestion
  "verifiableDataRegistry": "DID Specification"
```

Author: I think i got this in my lates changes.

}
```

> Please ensure your pull request and entry are passing the required validation rules.

Reviewer: ```suggestion
Please ensure your pull request and entry pass the required validation rules.
```

### Registration Review Process

Your pull request will be reviewed by the registry editors.

> Changes regarging the required criteria may be requested.

Author: ```suggestion
Changes regarding the required criteria may be requested.
```

Author: originally from: https://github.com/w3c/did-spec-registries/pull/356/files#r741205421

> If there are no objections or changes requested, 
> 
> your method will be registered after a minimum of 7 days.

Reviewer: ```suggestion
If there are no objections or changes requested, 
your method will be registered after a minimum of 7 days.
```

## Register a DID Method

### Required Registration Criteria

Please confirm:

- [ ] The DID Method specification defines the identifer syntax.
- [ ] The DID Method specification defines the supported DID Operations.
- [ ] The DID Method specification contains a Privacy and Security Considerations.

### Optional Registration Criteria

Consider ensuring:

- [ ] The registration entry contains contact information.
- [ ] The registration entry contains registry information.

### Registration Process

In order to register a New DID Method, you must add a json file to [./methods](./methods).

Here is an example registration entry:

```jsonc
{

>   // These fields are required

Reviewer 2: ```suggestion
  // These fields are required (and this line must be deleted, as JSON doesn't support comments)
```

Reviewer: I don't think we need to say this because 1) most people working w/ JSON know this and most registrants have a technical background, and 2) the schema validation will fail and will tell them why if they keep the comments in the file.

Reviewer 2: Is it possible to at least apply the clever CSS that hides the inline comments from a select&copy?  I have problems with including inline comments in examples of any language/syntax that doesn't really support those comments, no matter what ineffective fake-escape is applied (`//`, `--`, `#`, `/*`, `/!` etc.).  If comments are needed in that language (and I do agree that they're helpful here!), then that language needs a revision to support them (which I understand is at least under consideration for the next rev of JSON, but it isn't there yet).

Reviewer: > Is it possible to at least apply the clever CSS that hides the inline comments from a select&copy?

No, I don't think it is, because this is Markdown that's rendered in the Github interface (and they don't let you have control over CSS styles IIRC because that can be used as an attack/tracking vector). I'll look into it after merging this PR (I can't test this stuff out w/o it being live on the Github site since this is a PR template, IIRC).

Reviewer 2: Ah, my bad. I didn't look at the filename. :-/  I'll rejoice if GitHub Markdown someday supports such a thing, but I won't hold my breath waiting for it.

  "name": "example",
  "status": "registered",
  "specification": "https://w3c-ccg.github.io/did-spec/",
  // These fields are optional
  "contactName": "W3C Credentials Community Group",
  "contactEmail": "",

>   "contactWebsite": "",

Reviewer: This is for future PRs, just writing it down while I'm thinking about it:

We should reject registrations that don't provide a `contactEmail` address. It's a huge pain trying to figure out how to get in touch w/ these DID Method authors.

Author: we can very easily do that, now that we have: 

https://github.com/w3c/did-spec-registries/pull/361/files#diff-ea6b2d66de4a221273936ab35afb0fae0f8ace048d29642c2440dcfe59c206e1R6

You can request changes on that PR, to add these fields to the required section... however, they will still be empty for many existing entries....  I think its best we not co-mingle that issue with updating the process, we can address it after we have changed the process, adding it to the current list of changes will make these PRs more contentious and harder to review.

Author: I opened: https://github.com/w3c/did-spec-registries/issues/362 Please keep the discussion to that issue to keep the PR free from it.

Author: I'm resolving this thread so folks know to contribute to the issue regarding this conversation.

  "verifiableDataRegistry": "DID Specification"
}
```

Please ensure your pull request and entry pass the required validation rules.

### Registration Review Process

Your pull request will be reviewed by the registry editors.

Changes regarding the required criteria may be requested.

If there are no objections or changes requested, 

> your method will be registered after a minimum of 7 days.

Reviewer 3: we could waffle about it and say "but it could take 30 days or more."

that would set expectations without committing editors to a timeline.

> your method will be registered after a minimum of 7 days.

Reviewer 2: If the editors feel that such a waffle is required, I could perhaps be convinced.  I do feel that there should be *some* commitment to a maximum, by which point the submitter should expect a "registered" or "denied because ..." action, such that they can try to correct whatever failing was found.  "Rejection implied by silence" doesn't make me feel good.

> your method will be registered after a minimum of 7 days.

Reviewer 3: that is a good point

> your method will be registered after a minimum of 7 days.

Reviewer: I'm fine w/ max of 30 days. They can start loudly complaining at that point and I'd personally feel that it was justified.
